examples/hosted-wallet-kt/pfi/src/main/kotlin/api/Exchanges.kt

```diff
@@ -0,0 +1,140 @@
+package api
+
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.OkHttpClient
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Request as OkHttpRequest
+import spark.Request
+import spark.Response
+import spark.Spark.post
+import spark.Spark.put
+import tbdex.sdk.httpclient.CreateExchangeRequestBody
+import tbdex.sdk.messages.Quote
+import tbdex.sdk.messages.QuoteData
+import tbdex.sdk.messages.QuoteDetails
+import tbdex.sdk.messages.Order
+import tbdex.sdk.messages.OrderStatus
+import tbdex.sdk.messages.OrderStatusData
+import tbdex.sdk.messages.Close
+import tbdex.sdk.messages.CloseData
+import tbdex.sdk.web5.BearerDid
+
+class Exchanges(private val bearerDid: BearerDid, private val offeringsRepository: data.Offerings) {
+    init {
+        post("/exchanges") { req, res -> createExchange(req, res) }
+        put("/exchanges/:id") { req, res -> completeOrder(req, res) }
+    }
+
+    private var exchangesToReplyTo: MutableMap<String, String> = mutableMapOf()
+
+    private fun createExchange(req: Request, res: Response): String {
+        val requestBody = CreateExchangeRequestBody(req.body())
+
+        val replyTo = requestBody.replyTo ?: throw Exception("replyTo cannot be null for this example")
+        val rfq = requestBody.rfq
+
+        rfq.verifyOfferingRequirements(this.offeringsRepository.getOffering(rfq.data.offeringId))
+
+        this.exchangesToReplyTo[rfq.metadata.exchangeId] = replyTo
+
+        res.status(202)
+
+        Thread {
+            Thread.sleep(3000)
```

Comment:
should there be retries?

Reply:
maybe eventually, but it's. a bit much right now. prioritizing "happy path" to maximize ease of learning